src/requests/create-signature.ts

```diff
@@ -0,0 +1,112 @@
+import * as crypto from 'crypto'
+import type {
+  CanonicalRequest,
+  NormalizedCanonicalRequest,
+  NormalizedHeaders,
+  Secret,
+  Timestamp,
+} from './typings'
+import { CanonicalRequestValidator, SecretValidator, TimestampValidator } from './typings'
+import { getNormalizedEncodedURI, getNormalizedHeaders } from './utils'
+import { ContentfulSigningHeader } from './constants'
+
+const hash = (normalizedCanonicalRequest: NormalizedCanonicalRequest, secret: string) => {
+  const stringifiedHeaders = normalizedCanonicalRequest
+    .headers!.map(([key, value]) => `${key}:${value}`)
+    .join(';')
+
+  const stringifiedRequest = [
+    normalizedCanonicalRequest.method,
+    normalizedCanonicalRequest.path,
+    stringifiedHeaders,
+    normalizedCanonicalRequest.body,
+  ].join('\n')
+
+  const hmac = crypto.createHmac('sha256', secret)
+
+  hmac.update(stringifiedRequest)
+
+  return hmac.digest('hex')
+}
+
+const enrichNormalizedHeadersWithMetadata = (headers: NormalizedHeaders, timestamp: number) => {
+  const result = [...headers]
+  const headerKeys = headers.map(([key]) => key)
+  const joinedSignedHeaders = headerKeys
+    // We always sign timestamp
```

Comment:
What about x-contentful-signed-headers?

Reply:
That is actually included as well (see few lines below), but the reason for inclusion is that we always have a timestamp, that's what the comment is for. 
I just wanted to point out that you may not want to include additional headers to sign, and in that case the sole signed-header would be the timestamp, along with the signed headers.

is the comment confusing?